src/dscigametrics/metric4.py

```diff
@@ -0,0 +1,26 @@
+def find_best_and_worst_campaigns(df, start_date, end_date, campaign_ids, metric):
```

Comment:
Could you change your file name to function name? Appreciate it.

Reply:
All others looks good to me now.

Reply:
happy now?